### DISCUSSION TITLE
Ban a process from getting elected to leader

### DISCUSSION THREAD

**Discussion by daande** [sidekiq/sidekiq#5817] [Q&A]
Is it possible to ban a particular queue from being elected leader?

**Comment by mperham** [comment]
A queue is a data structure within Redis. A Sidekiq process is elected leader. And no, there is no way to prevent a specific process from being elected.

**Reply by daande** [reply]
`And no, there is no way to prevent a specific process from being elected.` 

@mperham Can I add this as a feature request?  We would like to deprioritize a specific process from being elected leader possibly through an environment variable.

**Reply by mperham** [reply]
You'll need to give a better explanation and justification. For years, I've told people that it is a mistake to specialize Sidekiq processes.

**Reply by daande** [reply]
We are running a sidekiq process literally named `unreliable` as it is running code which is known to segfault.  I am worried that our Periodic Jobs might not be queued when they should be if the `unreliable` process is elected leader and a segfault occurs.

**Reply by mperham** [reply]
That's a pretty good justification. 👍🏼

**Reply by mperham** [reply]
So in Enterprise 7.1, you'll be able to do this:

SIDEKIQ_LEADER=false bundle exec sidekiq -q unreliable

The only caveat is that you must be careful not to mark all processes this way or you'll get no leader.

```
% SIDEKIQ_LEADER=false bundle exec sidekiq   
...[snip]...
2023-03-08T17:19:17.687Z pid=11402 tid=6he WARN: This Sidekiq Enterprise process cannot become leader
2023-03-08T17:19:17.687Z pid=11402 tid=6he INFO: Starting processing, hit Ctrl-C to stop
2023-03-08T17:19:17.687Z pid=11402 tid=1ce INFO: SuperFetch[default] activated
```

**Reply by daande** [reply]
Thank you @mperham I appreciate this, we are currently running 16 sidekiq processes in production and only 3 are for the unreliable process.

**Reply by daande** [reply]
😨 
![image](https://user-images.githubusercontent.com/3680231/223798814-7fd14802-8265-4597-9731-ab3bb4b18ef7.png)

**Reply by mperham** [reply]
7.1 has not been released yet. You can manually shut down that instance to trigger a reelection. Your systemd or PaaS should restart the process.

**Reply by daande** [reply]
Yes, I know was just sharing what spurred this conversation!
